motionplan/plan_test.go

```diff
@@ -0,0 +1,227 @@
+package motionplan
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+	pb "go.viam.com/api/service/motion/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/motionplan/ik"
+	"go.viam.com/rdk/motionplan/tpspace"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+)
+
+func TestEvaluateTrajectory(t *testing.T) {
+	plan := Trajectory{
+		map[string][]referenceframe.Input{"": {{1.}, {2.}, {3.}}},
+		map[string][]referenceframe.Input{"": {{1.}, {2.}, {3.}}},
+	}
+	// Test no change
+	score := plan.EvaluateCost(ik.L2InputMetric)
+	test.That(t, score, test.ShouldEqual, 0)
+
+	// Test L2 for "", and nothing for plan with only one entry
+	plan = append(plan, map[string][]referenceframe.Input{"": {{4.}, {5.}, {6.}}, "test": {{2.}, {3.}, {4.}}})
+	score = plan.EvaluateCost(ik.L2InputMetric)
+	test.That(t, score, test.ShouldEqual, math.Sqrt(27))
```

Comment:
Given these are floats should this be using `AlmostEqual`?

Reply:
Yeah idk how this is even passing tbh